scripts/tank_cmd.py

```diff
@@ -379,35 +396,162 @@ def shotgun_cache_actions(log, pipeline_config_root, args):
         log.info("")
 
 
-def shotgun_run_action(log, install_root, pipeline_config_root, is_localized, args):
+def shotgun_run_action_auth(log, install_root, pipeline_config_root, is_localized, args):
     """
-    Executes the special shotgun run action command from inside of shotgun
+    Executes the special shotgun run action command from inside of shotgun.
+    Authenticated version. 
+    
+    This method expects the following arguments to be passed via the args param:
+    args[0]: name of the action
+    args[1]: entity type to operate on
+    args[2]: list of entity ids as a string, e.g. '1,2,3'
+    args[3]: shotgun user login requesting this command
+    args[4]: password for the user. Can be set to "" - in that case,
+             this is a hint to toolkit to try to authenticate via a
+             cached session token
+    args[5:]: reserved for future use. This method will *not* error 
+              if unexpected args are passed to it.
+    
+    :param log: Python logger
+    :param install_root: Root of the toolkit core installation
+    :param pipeline_config_root: Root of the pipeline configuration
+    :param is_localized: True if the pipeline configuration has been localized
+    :param args: list of arguments passed from Shotgun.
     """
-
     # we are talking to shotgun! First of all, make sure we switch on our html style logging
     log.handlers[0].formatter.enable_html_mode()
 
+    log.debug("Running shotgun_run_action_auth command")
+
+    action_name = args[0]
+    entity_type = args[1]
+    entity_ids_str = args[2].split(",")
+    entity_ids = [int(x) for x in entity_ids_str]
+    login = args[3]
+    password = args[4]
+
+    # now, first try to authenticate
+    dm = DefaultsManager()
+    sa = ShotgunAuthenticator(dm)
+
+    # first of all, if there is a default user defined, that takes precedence
+    # TODO: this interface call is likely to change.
```

Comment:
Hum... if you don't have a saved user, this will prompt you for your credentials. I'm surprised this worked on your end.